### ISSUE TITLE
Take overlaps into account when resizing and click-dragging the time entry

### ISSUE THREAD

**Issue by skel35** [toggl-open-source/toggldesktop#4623]
### 💻 Environment
Platform: Windows

### 📒 Description

When resizing the time entry, the preview doesn't take overlaps into account. The gap buttons should also not be shown in the place of the time entry preview.
![4jYeh1e4cb](https://user-images.githubusercontent.com/16451391/97555401-c09fd100-19e0-11eb-97e9-78f358390e55.gif)

On macOS the overlaps are already being taken into account when resizing, drag&dropping, or click-dragging the time entry.
This creates a more pleasant user experience.

**Comment by atsvirchkova** [comment]
From what I see on master MacOS, it currently works pretty much like Windows. Or am I looking in the wrong place?
![resizing](https://user-images.githubusercontent.com/11901598/97981062-689b0d00-1dda-11eb-9297-7de33902e0bd.gif)
Should it hide description and change horizontal offset of TEs during resizing (like it does after resizing)?

**Comment by skel35** [comment]
You're right, the overlaps on macOS are only taken into account during click-dragging, not resizing or drag&dropping.
